### PR TITLE
Introduces demo project in Kotlin

### PR DESCRIPTION
**Pull request by babedev**
Convert demo project from Java to Kotlin.

### CODE REVIEW

File: demo-kotlin/src/main/java/com/bluelinelabs/conductor/demo/kotlin/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class MainActivity : AppCompatActivity() {
+
+    private var router: Router? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        setSupportActionBar(toolbar)
+
+        controller_container
```
Comment (Judrummer) [inline demo-kotlin/src/main/java/com/bluelinelabs/conductor/demo/kotlin/MainActivity.kt]: remove this line

File: demo-kotlin/src/main/java/com/bluelinelabs/conductor/demo/kotlin/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import android.support.v7.app.AppCompatActivity
+import com.bluelinelabs.conductor.Conductor
+import com.bluelinelabs.conductor.Router
+import com.bluelinelabs.conductor.RouterTransaction
+import com.bluelinelabs.conductor.demo.kotlin.controllers.HomeController
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+    private var router: Router? = null
```
Comment (Judrummer) [inline demo-kotlin/src/main/java/com/bluelinelabs/conductor/demo/kotlin/MainActivity.kt]: better to use lateinit or lazy

**Comment by PaulWoitaschek** [comment]
There are no kotlin specific features in conductor.
I don't think we need a kotlin demo project. (demo projects also need to maintained).

**Comment by hijamoya** [comment]
I have a demo project 
https://github.com/hijamoya/Android-Kotlin-Dagger2-Conductor-Example

**Pull request closed without merging**